packages/js/src/ui/components/Renderer.tsx

```diff
@@ -67,33 +68,40 @@ type RendererProps = {
   preferencesFilter?: PreferencesFilter;
   routerPush?: RouterPush;
   novu?: Novu;
+  containerElement?: Node | null | undefined;
 };
 
 export const Renderer = (props: RendererProps) => {
   const nodes = () => [...props.nodes.keys()];
 
   onMount(() => {
-    const id = 'novu-default-css';
-    const el = document.getElementById(id);
+    const id = NOVU_DEFAULT_CSS_ID;
+    const root = props.containerElement instanceof ShadowRoot ? props.containerElement : document;
+    const el = root.getElementById(id);
     if (el) {
       return;
     }
 
     const styleEl = document.createElement('style');
     styleEl.id = id;
-    document.head.insertBefore(styleEl, document.head.firstChild);
     styleEl.innerHTML = css;
 
+    const stylesContainer = props.containerElement ?? document.head;
+    stylesContainer.insertBefore(styleEl, stylesContainer.firstChild);
```

Comment:
inject the default styles to the `containerElement` if provided otherwise document head